### PR TITLE
Refresh scope beans

### LINKED ISSUE #2264: Feature request: Aggregation store - reload models in Hjson config files at runtime without restarting application
Currently any model change in Hjson config files does require an application restart so that the change will be effective. Possibility to reload models in Hjson config files at runtime would make analytic features more appealing and convenient for Elide users.

### PR DESCRIPTION
**Pull request by aklish**
Resolves #2264 
## Description

Makes spring controllers and Elide beans `@RefreshScope` so they will automatically reload when a spring boot based Elide service (with actuators enabled) is refreshed (through the actuator /refresh endpoint).

## Motivation and Context
To enable an HJSON editor that can save and automatically update dynamic models.

## How Has This Been Tested?
Existing tests plus manual tests.

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: elide-spring/elide-spring-boot-autoconfigure/src/test/java/example/tests/HeaderCleansingTest.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2019, Yahoo Inc.
```
Comment (jkusa) [inline elide-spring/elide-spring-boot-autoconfigure/src/test/java/example/tests/HeaderCleansingTest.java:2]:
```suggestion
 * Copyright 2021, Yahoo Inc.
```

**Review by jkusa — APPROVED** [review]

File: elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideAutoConfiguration.java [hunk 2]
```diff
@@ -127,17 +140,34 @@ public DataSourceConfiguration getDataSourceConfiguration() {
...
      * Creates the Elide instance with standard settings.
      * @param dictionary Stores the static metadata about Elide models.
      * @param dataStore The persistence store.
+     * @param transactionRegistry Global transaction registry.
      * @param settings Elide settings.
      * @return A new elide instance.
      */
     @Bean
     @ConditionalOnMissingBean
+    @Scope(SCOPE_PROTOTYPE)
```
Comment (thaingo) [inline elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideAutoConfiguration.java:167]: Should we clean up prototype scoped objects? It is recommended [here](https://docs.spring.io/spring-framework/docs/3.0.0.M3/reference/html/ch04s04.html#beans-factory-scopes-prototype)

Comment (aklish) [reply elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideAutoConfiguration.java:167]: Good call, but I think I might fold these beans into the the Refresh versions.  I think it is confusing having both.  They are essentially private.

Comment (aklish) [reply elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideAutoConfiguration.java]: I did a small refactor.  For the beans that are still declared PROTOTYPE scope, I don't believe any cleanup is required as they don't hold onto connections or sessions.  The only thing which manages connections are transactions.  We expect transactions to complete (or timeout) after which their resources should be cleaned up.  The controllers are also RefreshScope - so they should be garbage collected along with their associated references to any prototype beans.

**Review by moizarafat — APPROVED** [review]

**Pull request merged**